homeassistant/components/statistics/sensor.py

```diff
@@ -162,70 +169,81 @@ def valid_binary_characteristic_configuration(config):
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
     """Set up the Statistics sensor."""
 
     await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
 
     async_add_entities(
-        [
+        new_entities=[
             StatisticsSensor(
-                source_entity_id=config.get(CONF_ENTITY_ID),
-                name=config.get(CONF_NAME),
+                source_entity_id=str(config.get(CONF_ENTITY_ID)),
+                name=str(config.get(CONF_NAME)),
                 unique_id=config.get(CONF_UNIQUE_ID),
-                state_characteristic=config.get(CONF_STATE_CHARACTERISTIC),
-                samples_max_buffer_size=config.get(CONF_SAMPLES_MAX_BUFFER_SIZE),
+                state_characteristic=str(config.get(CONF_STATE_CHARACTERISTIC)),
+                samples_max_buffer_size=cast(
+                    int, config.get(CONF_SAMPLES_MAX_BUFFER_SIZE)
+                ),
                 samples_max_age=config.get(CONF_MAX_AGE),
-                precision=config.get(CONF_PRECISION),
-                quantile_intervals=config.get(CONF_QUANTILE_INTERVALS),
-                quantile_method=config.get(CONF_QUANTILE_METHOD),
+                precision=cast(int, config.get(CONF_PRECISION)),
+                quantile_intervals=cast(int, config.get(CONF_QUANTILE_INTERVALS)),
+                quantile_method=str(config.get(CONF_QUANTILE_METHOD)),
             )
         ],
-        True,
+        update_before_add=True,
     )
-    return True
 
 
 class StatisticsSensor(SensorEntity):
     """Representation of a Statistics sensor."""
 
     def __init__(
         self,
-        source_entity_id,
-        name,
-        unique_id,
-        state_characteristic,
-        samples_max_buffer_size,
-        samples_max_age,
-        precision,
-        quantile_intervals,
-        quantile_method,
-    ):
+        source_entity_id: str,
+        name: str,
+        unique_id: str | None,
+        state_characteristic: str,
+        samples_max_buffer_size: int,
+        samples_max_age: timedelta | None,
+        precision: int,
+        quantile_intervals: int,
+        quantile_method: str,
+    ) -> None:
         """Initialize the Statistics sensor."""
-        self._source_entity_id = source_entity_id
-        self.is_binary = self._source_entity_id.split(".")[0] == "binary_sensor"
-        self._name = name
-        self._unique_id = unique_id
-        self._state_characteristic = state_characteristic
+        self._attr_icon: str = ICON
+        self._attr_name: str = name
+        self._attr_should_poll: bool = False
+        self._attr_unique_id: str | None = unique_id
+        self._source_entity_id: str = source_entity_id
+        self.is_binary: bool = (
+            self._source_entity_id.split(".", maxsplit=1)[0] == "binary_sensor"
```

Comment:
Following a pylint warning. Are there any entity ID helpers instead of using `str`?

Reply:
Yeah, use `split_entity_id` from `homeassistant.core`.
Also, we should import `DOMAIN` from `binary_sensor` as `BINARY_SENSOR_DOMAIN` and compare against that.